sapientml_core/templates/model_templates/model_train.py.jinja

```diff
@@ -17,25 +17,47 @@ model = {{ model_name }}(random_state=random_state_model, {{ params }})
 from sklearn.multioutput import MultiOutputRegressor
 
 model = MultiOutputRegressor(model)
+
 {% elif is_multioutput_classification %}
 from sklearn.multioutput import MultiOutputClassifier
 
 model = MultiOutputClassifier(model)
 {% endif %}
+
 {% set xgbclassifier = "XGBClassifier" %}
-{% if model_name == xgbclassifier %}
+
+{% if pipeline.task.target_columns|length == 1 %}
 from sklearn.preprocessing import LabelEncoder
+flag=0
+if target_train.select_dtypes(include=['object']).columns.any():
+   str_columns = target_train.select_dtypes(include=['object']).columns
+   label_encoder= LabelEncoder()
+   flag=1
+   for col in str_columns:
+       target_train[col] = label_encoder.fit_transform(target_train[col])
 
-label_encoder = LabelEncoder()
-target_train = pd.DataFrame(label_encoder.fit_transform(target_train), columns=TARGET_COLUMNS)
-with open('target_LabelEncoder.pkl', 'wb') as f:
-    pickle.dump(label_encoder, f)
+if flag==1:
+    with open('target_LabelEncoder.pkl', 'wb') as f:
+        pickle.dump(label_encoder, f)
+    flag=0
 
-{% endif %}
-{% if pipeline.task.target_columns|length == 1 %}
 model.fit(feature_train, target_train.values.ravel())
 {% else %}
+from sklearn.preprocessing import LabelEncoder
+flag=0
+if target_train.select_dtypes(include=['object']).columns.any():
+   str_columns = target_train.select_dtypes(include=['object']).columns
+   label_encoder= LabelEncoder()
+   flag=1
+   for col in str_columns:
+       target_train[col] = label_encoder.fit_transform(target_train[col])
+
+if flag==1:
+    with open('target_LabelEncoder.pkl', 'wb') as f:
+        pickle.dump(label_encoder, f)
+    flag=0
```

Comment:
It is not preferable to show the `flag` variable to users. An implementation example from me is:
```
{{ % if categorical_target_columns }}
categorical_target_column = {{categorical_target_columns}} # passed from pipeline_template.py
label_encoders = {}
for col in categorical_target_columns:
    label_encoder = LabelEncoder()
    target_train[col] = label_encoder.fit_transform(target_train[col])

with open('target_LabelEncoder.pkl', 'wb') as f:
        pickle.dump(label_encoders, f)
{{ % endif }}
```